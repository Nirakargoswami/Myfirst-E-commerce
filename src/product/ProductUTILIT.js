import { createSelector } from "reselect";

const CARTSELECTOR = (state) => state.CARTDATA;

export const CARTITEM = createSelector([CARTSELECTOR], (item) => item.CARTITEM);

export const NO_OFITEM = createSelector([CARTITEM], (item) =>
item.reduce((x,y) => (x + y.Quantity),0)
);

export const TOTALINCART = createSelector(
  [CARTITEM],
  (item) =>  item.map((item) => item )

);
