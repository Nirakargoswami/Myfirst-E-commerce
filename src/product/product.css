* {
  font-family: Arial, Helvetica, sans-serif;
  box-sizing: border-box;
}
.Product_Main {
  display: block;
}


.CHECK{
    border: 1px black solid;
    color: red;
}