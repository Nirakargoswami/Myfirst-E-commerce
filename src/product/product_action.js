export const GETDATA = (payload) => {
  return {
    type: "Get_data",
    payload: payload,
  };
};

export const ADDTOCART = (Payload) => {
  return {
    type: "ADDTOCART",
    Payload: Payload,
  };
};
export const BUYNOW = (Payload) => {
  return {
    type: "BUYNOW",
    Payload: Payload,
  };
};
export const CLEANUP = (Payload) => {
  return {
    type: "CLEANUP",
    
    Payload: Payload,
  };
};