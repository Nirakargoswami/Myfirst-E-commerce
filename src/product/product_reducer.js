import DATA from "../alldata/alldata";
import { utiliti } from "../product/Productheader/utiliti";
import  Buyitem  from "../product/Productheader/ONEMORE";
 export const Intialstate = {
  DATA: [],
  CARTITEM: [],
};
const Product_DATA = (state = Intialstate, action) => {
  switch (action.type) {
    case "Get_data":
      return { ...state, DATA: DATA };
    case "ADDTOCART":
      return {
        ...state,
        CARTITEM: utiliti(action.Payload, state.CARTITEM,state.DATA),
      };
    case "BUYNOW":
      return {
        ...state,
        CARTITEM: Buyitem(action.Payload),
      };
    case "CLEANUP": {
      return {
        ...state,
        CARTITEM: [],
      };
    }
    default:
      return state;
  }
  
};
export default Product_DATA;
