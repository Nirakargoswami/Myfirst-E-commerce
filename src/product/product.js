import React, { Component } from "react";

import "/home/bhumika/nirakar/travell-app/src/allcomponent/header/headre.css";
import "/home/bhumika/nirakar/travell-app/src/product/product.css";
import Prouctartical from "./ProductArtical/artical";
import {BUTTON}  from "../allcomponent/newcomonent"
export default class Productpage extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    console.log("history fri the me "+this.props);
    return (
      <div>
        <div className="Product_Main">
          <div>
            <Prouctartical NEWDATA={this.props.match.params.productName} />
            <div>
              <BUTTON
                style={{ color: "black" }}
                onClick={() => this.props.history.push("/")}
                type="button"
              >
                <h5>BACK to home</h5>
              </BUTTON>
            </div>
          </div>

          {/* od not touch under this line  */}
        </div>
      </div>
    );
  }
}
