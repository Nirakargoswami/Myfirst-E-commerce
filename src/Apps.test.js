import React from "react";
import { shallow, mount } from "enzyme";

import {BUYNOW} from "./product/product_action"
import NEW from "./Propscomponetn"

import Product_DATA from "./product/product_reducer"






let COSNTStent;


beforeEach(() => {

  const PROPS = {props:{color:"black"}}; 
  COSNTStent = mount(<NEW props={PROPS} />);

})


  it("lets see  function is working or not " , () => {

    
     let tree = COSNTStent
     expect(tree).toMatchSnapshot();


  }


  )

   it("lets see  function is working or not ", () => {


     let tree = COSNTStent;
     expect(tree).toMatchSnapshot();
   });


   it("lets see  function is working  ", () => {
     const PROPS = { props: { color: "black" } }; 
     expect(shallow(<NEW props={PROPS} />).state()).toEqual({ color: "black" });
   });




const intialstate = {
  
  DATA: [],
  CARTITEM: [],

}
   it("lets check reducer test wrolimng or not " , () => {
     console.log(Product_DATA)
     console.log(BUYNOW)
   expect(
     Product_DATA(intialstate, {
       type: "BUYNOW",
      Payload : ""
     })
   ).toEqual({
     DATA: [],
     CARTITEM: [],
     ispadimg: "true"
   });
   }
    
   )

     

   it("lets check action is working  not ",() => {
const  Payload = "Nirakar"
    const Actiondata = {
      type: "BUYNOW",
      Payload: Payload
    };
  
 expect(BUYNOW(Payload)).toEqual(Actiondata)

   })