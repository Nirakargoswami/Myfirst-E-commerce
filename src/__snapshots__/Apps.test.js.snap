// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lets see  function is working or not  1`] = `ReactWrapper {}`;

exports[`lets see  function is working or not  2`] = `ReactWrapper {}`;
